### PR TITLE
Add ability to get auth tokens from GCP.

### PR DESCRIPTION
**Pull request by kingsukhoi**
This is to enable the ability to connect to Cloudsql using IAM accounts

### CODE REVIEW

**Comment by onlywicked** [comment]
There is a problem.

To natively use CloudSQL, you have to use `db/sql`. Some orm does not support it.

**Side note:**
All Google Client Libraries uses ADC for loading credentials.
So, this does not serve any purpose, as far as I can see.

**Review by onlywicked — COMMENTED** [review]
There is a problem.

To natively use CloudSQL, you have to use `db/sql`. Some orm does not support it.

**Side note:**
All Google Client Libraries uses ADC for loading credentials.
So, this does not serve any purpose, as far as I can see.

File: GCPAuthTokens/token.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const (
+	EnvServiceAcctFile = "GOOGLE_APPLICATION_CREDENTIALS"
+)
+
+func isEnvExist(key string) bool {
+	if _, ok := os.LookupEnv(key); ok {
+		return true
+	}
+	return false
+}
```
Comment (onlywicked) [inline GCPAuthTokens/token.go:21]: I don't think we have to branch here. `ok` is already a boolean so, we can return it directly.

But I don't think we need this function.
We can directly use `os.Getenv` or `os.LookupEnv` and do the branching directly there.

Comment (kingsukhoi) [reply GCPAuthTokens/token.go]: Good call
Fixed now

File: GCPAuthTokens/token.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	token, err := c.GenerateAccessToken(ctx, requestOpts)
+
+	if err != nil {
+		return "", err
+	}
+
+	return token.AccessToken, nil
+}
```
Comment (onlywicked) [inline GCPAuthTokens/token.go:77]: This too functions are literally the same.
So, please they can easily encapsulated.

**Review by onlywicked — CHANGES_REQUESTED** [review]

File: gcpauth/token.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	return getToken(ctx, saEmail, credentialsClient)
+}
+
+func GetAuthFromFile(path, saEmail string) (string, error) {
+	ctx := context.Background()
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return "", err
+	} else if err != nil {
+		return "", err
+	}
```
Comment (onlywicked) [inline gcpauth/token.go:48]: Just return the error directly. `os.IsNotExist(err)` is not serving any purpose here.

**Review by onlywicked — CHANGES_REQUESTED** [review]

**Comment by kingsukhoi** [comment]
https://linear.app/flahmingo/issue/DEVOP-13/dig-deeper-into-permissions-on-the-database

**Comment by kingsukhoi** [comment]
https://linear.app/flahmingo/issue/DEVOP-38/secrets-library

**Review by onlywicked — APPROVED** [review]

**Pull request merged**